pkg/virt-api/rest/subresource.go

```diff
@@ -1545,3 +1555,261 @@ func (app *SubresourceAPIApp) RemoveMemoryDumpVMRequestHandler(request *restful.
 
 	response.WriteHeader(http.StatusAccepted)
 }
+
+func generateVMIInterfaceRequestPatch(vmi *v1.VirtualMachineInstance, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
```

Comment:
Not a must, but this seems to fit the existing `vmispec` package.
You could just use the [filter function](https://github.com/kubevirt/kubevirt/blob/2f84370ea7c08547520776f2ccb934cec8257ea5/pkg/network/vmispec/interface.go#L51), or create a lookup one dedicated per name.

The it will look like this:
```
ifaces := netvmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(i v1.Interface) bool { return i.name == dynamicIfaceName(interfaceRequest)})
if (iface) > 0 { return "", error...}
```